### DISCUSSION TITLE
How to configure client request timeout so that it can fail fast and retry for DynamoDB operations

### DISCUSSION THREAD

**Discussion by adeshg** [aws/aws-sdk-go-v2#2180] [Q&A]
I am using this SDK for DynamoDB operations in my application.
We have seen occasional spikes in latency which causes requests to timeout.
I want to configure the client in such a way that the request times out fast and automatically retries with exponential backoff.
AWS support pointed me to this resource https://aws.amazon.com/blogs/database/tuning-aws-java-sdk-http-request-settings-for-latency-aware-amazon-dynamodb-applications/ , but it looks like that is for Java SDK and I am not sure how to map that for the go sdk.

Any help/examples and insight would be great. Thanks

**Comment by jmklix** [comment]
You should try using [WithTimeout](https://pkg.go.dev/context#WithTimeout) for this.
```go
func slowOperationWithTimeout(ctx context.Context) (Result, error) {
	ctx, cancel := context.WithTimeout(ctx, 100*time.Millisecond)
	defer cancel()  // releases resources if slowOperation completes before timeout elapses
	return slowOperation(ctx)
}
```
